crates/biome_js_analyze/src/lint/nursery/no_array_constructor.rs

```diff
@@ -0,0 +1,99 @@
+    ///
+    /// Use of the Array constructor to construct a new array is generally discouraged in favor of array literal notation because of the single-argument pitfall and because the Array global may be redefined.
+    /// The exception is when the Array constructor is used to intentionally create sparse arrays of a specified size by giving the constructor a single numeric argument.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// Array(0, 1, 1);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// Array(500);
+    /// [0, 1, 2];
+    /// ```
+    ///
+    pub NoArrayConstructor {
+        version: "next",
+        name: "noArrayConstructor",
+        sources: &[RuleSource::Eslint("no-array-constructor")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for NoArrayConstructor {
+    type Query = Ast<JsExpressionStatement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let expression_statement = ctx.query();
+
+        let expression = expression_statement.expression().ok()?;
+
+        match expression {
+            AnyJsExpression::JsCallExpression(call_expression) => {
+                let callee = call_expression.callee().ok()?;
+                let arguments = call_expression.arguments().ok()?;
+                validate(&callee, &arguments)
+            }
+            AnyJsExpression::JsNewExpression(new_expression) => {
+                let callee = new_expression.callee().ok()?;
+                let arguments = new_expression.arguments()?;
+                validate(&callee, &arguments)
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Don't use Array constructors."
+                },
+            )
+            .note(markup! {
+                "Use of the Array constructor is not allowed except creating sparse arrays of a specified size by giving a single numeric argument."
+            })
+            .note(markup! {
+                "The array literal notation [] is preferable."
+            }),
+        )
+    }
+}
+
+fn validate(callee: &AnyJsExpression, arguments: &JsCallArguments) -> Option<()> {
+    let mut args_iter = arguments.args().into_iter();
+    let first_arg = args_iter.next();
+    let second_arg = args_iter.next();
+    if callee.text() == "Array" {
+        if first_arg.is_some()
+            && second_arg.is_none()
+            && !matches!(first_arg?.ok()?, AnyJsCallArgument::JsSpread(_))
+        {
+            return None;
+        }
+        Some(())
+    } else {
+        None
+    }
+}
```

Comment:
nits; we can use early return here.
```suggestion
fn validate(callee: &AnyJsExpression, arguments: &JsCallArguments) -> Option<()> {
    if callee.text() != "Array" {
        return None
    }
    let mut args_iter = arguments.args().into_iter();
    let first_arg = args_iter.next();
    let second_arg = args_iter.next();
    if first_arg.is_some()
        && second_arg.is_none()
        && !matches!(first_arg?.ok()?, AnyJsCallArgument::JsSpread(_))
    {
        return None;
    }
    Some(())
}
```

Reply:
That't true, thank you...!